src/client/graphics/layers/EventsDisplay.ts

```diff
@@ -823,17 +825,54 @@ export class EventsDisplay extends LitElement implements Layer {
     `;
   }
 
+  private calculateBoatCountdown(boat: UnitView): number {
+    const currentTick = this.game.ticks();
+    const createdAt = boat.createdAt();
+    const ticksTraveled = currentTick - createdAt;
+
+    // Try to use server-provided estimated arrival tick if available
+    const estimatedArrivalTick = boat.estimatedArrivalTick();
+
+    if (estimatedArrivalTick !== undefined) {
+      const remaining = Math.max(0, estimatedArrivalTick - currentTick);
+      return remaining;
+    }
+
+    // If no server estimate available, show "Calculating..." status
+    return -1; // Special value to indicate "Calculating..." status
+  }
+
+  private formatCountdown(ticks: number): string {
+    if (ticks === -1) {
+      return translateText("events_display.boat_countdown.calculating");
+    }
+
+    if (ticks <= 0) {
+      return translateText("events_display.boat_countdown.arriving");
```

Comment:
Since this shows for less than a second it feels awkard since you can't read it.
It also seems to flicker between this and 1s. (evanpelle has called out using Pathfinder/a* instead)

I'd instead make the floor since arriving is assumed, unlike calculating.

    if (ticks <= 0) {
      return